server/meta.jsx

```diff
@@ -90,58 +90,44 @@ function validateWebSDKUrl({ pathname, query }) {
       `Invalid path for web-sdk bridge url: ${pathname || "undefined"}`
     );
   }
-  // check for extraneous parameters
-  const validWebSDKBridgeParams = [
-    "origin",
-    "version",
-    "payment-flow",
-    "debug",
-  ];
-  for (const param of Object.keys(query)) {
-    if (!validWebSDKBridgeParams.includes(param)) {
-      throw new Error(`Invalid parameter on web-sdk bridge url: ${param}`);
-    }
-  }
+
+  // supported query string parameters
+  const { origin, version, ["payment-flow"]: paymentflow, debug } = query;
 
   // validate the version parameter
-  if (query.version === undefined || !semverRegex.test(query.version)) {
+  if (version === undefined || !semverRegex.test(version)) {
     throw new Error(
-      `Invalid version parameter on web-sdk bridge url: ${query.version}`
+      `Invalid version parameter on web-sdk bridge url: ${version}`
     );
   }
 
   // validate the payment-flow parameter
   const validPaymentFlows = ["popup", "modal", "payment-handler"];
-  if (
-    query["payment-flow"] === undefined ||
-    !validPaymentFlows.includes(query["payment-flow"])
-  ) {
+  if (paymentflow === undefined || !validPaymentFlows.includes(paymentflow)) {
```

Comment:
Would it be better to do this validation on the server-side as well?

Reply:
I'm open to removing it here.